### PR TITLE
feat: add `rustls-tls`, `native-tls` features to enable equivalent features in `ureq` and `reqwest`

### LINKED ISSUE #110: Does not build for target `x86_64-unknown-linux-musl`
Building for the target `x86_64-unknown-linux-musl` does not work by default.

```bash
cargo build --target=x86_64-unknown-linux-musl
```

Workaround: disable the `reqwest` and `ureq` features.

### PR DESCRIPTION
**Pull request by dyc3**
fixes #110

#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,8 +31,15 @@ serde_json = "1.0"
 [dependencies.reqwest]
 version = "0.11"
 optional = true
+default-features = false
```
[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml:34]: It's to disable the `native-tls-crate` feature.

[the remark this answers is not in the document]
Comment (mgrachev) [reply Cargo.toml:34]: As far as I understand, this needs to be disabled only for the `x86_64-unknown-linux-musl` target, right?

[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml:34]: Yeah, you're right, but removing this line causes the build to fail for other targets. It appears that the target specific dependency doesn't override the dependency as one would expect. I'll have to look into this a bit more.

[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml]: I think I've figured it out with a better target condition.

[the remark this answers is not in the document]
Comment (mgrachev) [reply Cargo.toml]: I would go the other way - I would add 2 features to `update-informer`:
* `rustls-tls` (by default)
* `native-tls`

And I would turn on them on demand in the `ureq` and `request` crates.
What do you think?

[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml]: That would probably be better, but it's not immediately clear to me how we would do that.

According to [these docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html), specifying dependencies using `cfg(feature = "x")` does not work.

[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml]: Ah, apparently you can rename dependencies. I'll give it another go.

[the remark this answers is not in the document]
Comment (mgrachev) [reply Cargo.toml]: I think we can do that simplier:
```toml
# Cargo.toml
[features]
default = ["crates", "ureq", "rustls-tls"]
# ...
rustls-tls = ["ureq?/tls", "reqwest?/rustls-tls"]
native-tls = ["ureq?/native-tls", "reqwest?/native-tls"]
```

[the remark this answers is not in the document]
Comment (dyc3) [reply Cargo.toml]: Oh, idk how i missed that in the docs. However, that doesn't seem to be working.

File: Cargo.toml [hunk 2]
```diff
@@ -31,6 +33,7 @@ serde_json = "1.0"
...
 features = ["blocking", "json"]
 
 [dependencies.ureq]
```
Comment (mgrachev) [inline Cargo.toml:39]: Please add `default-features = false` and `features = ["gzip", "json"]` for the `ureq` client

**Comment by mgrachev** [comment]
Also, please add information about new features in the README.

File: README.md [hunk 3]
```diff
@@ -72,14 +72,16 @@ update-informer = { version = "1.0", default_features = false, features = ["gith
...
+| Name       | Type                | Default? |
+| ---------- | ------------------- | -------- |
+| crates     | Registry            | Yes      |
+| github     | Registry            | No       |
+| npm        | Registry            | No       |
+| pypi       | Registry            | No       |
+| [ureq]     | HTTP client         | Yes      |
+| [reqwest]  | HTTP client         | No       |
+| rustls-tls | HTTP client feature | Yes       |
+| native-tls | HTTP client feature | Yes       |
```
Comment (mgrachev) [inline README.md:84]:
```suggestion
| native-tls | HTTP client feature | No       |
```

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@dyc3 Thank you for your help 👍

**Pull request merged**